lib/url_for.js

```diff
@@ -16,7 +15,7 @@ function urlForHelper(path = '/', options) {
   const data = new URL(path, `http://${sitehost}`);
 
   // Exit if this is an external link
-  if (isExternalLink.call(this, path)) return path;
+  if (data.origin !== 'null' && data.hostname !== sitehost) return path;
```

Comment:
May accidentally fix #122?

Maybe add a test case?

Reply:
#122 requires the reverse, exit if `data.origin === 'null'.

I'll fix #122 in a separate PR (after this), #122 requires dropping support for [path with semicolon](https://github.com/hexojs/hexo-util/blob/master/test/url_for.spec.js#L76-L81).